File: src/main/java/com/back/domain/auction/auction/dto/response/AuctionPageResponse.java
package com.back.domain.auction.auction.dto.response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.data.domain.Page;

import java.util.List;

@Getter
@AllArgsConstructor
public class AuctionPageResponse {
    private List<AuctionListItemDto> content;
    private int page;
    private int size;
    private long totalElements;
    private int totalPages;

    public static AuctionPageResponse from(Page<AuctionListItemDto> page) {
        return new AuctionPageResponse(
                page.getContent(),
                page.getNumber(),
                page.getSize(),
                page.getTotalElements(),
                page.getTotalPages()
        );
    }
}

File: src/main/java/com/back/domain/auction/auction/dto/response/AuctionPageResponse.kt
package com.back.domain.auction.auction.dto.response

import org.springframework.data.domain.Page

data class AuctionPageResponse(
    val content: List<AuctionListItemDto>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int
) {
    companion object {
        @JvmStatic
        fun from(page: Page<AuctionListItemDto>): AuctionPageResponse =
            AuctionPageResponse(
                content = page.content,
                page = page.number,
                size = page.size,
                totalElements = page.totalElements,
                totalPages = page.totalPages
            )

        @JvmStatic
        fun of(
            content: List<AuctionListItemDto>,
            page: Int,
            size: Int,
            totalElements: Long
        ): AuctionPageResponse {
            val totalPages = if (totalElements == 0L) 0 else ((totalElements + size - 1) / size).toInt()
            return AuctionPageResponse(
                content = content,
                page = page,
                size = size,
                totalElements = totalElements,
                totalPages = totalPages
            )
        }
    }
}
